react/components/MiniCartContent.js

```diff
@@ -67,43 +71,25 @@ class MiniCartContent extends Component {
     this.sumItemsPrice(items) - totalPrice
 
   handleItemRemoval = async id => {
-    this.updateItemLoad(id, true)
-
     const {
-      data: orderFormContext,
-      data: { orderForm, updateAndRefetchOrderForm },
-      intl,
+      orderForm,
+      updateItems,
     } = this.props
     const itemPayload = orderForm.items.find(item => item.id === id)
     const index = orderForm.items.indexOf(itemPayload)
-    const updatedItem = [itemPayload].map(item => {
-      return {
-        id: parseInt(item.id),
-        index: index,
-        quantity: 0,
-        seller: 1,
-      }
-    })
+    const updatedItems = [itemPayload].map(item => ({
+      ...item,
+      index,
+      quantity: 0,
+      seller: 1,
+    }))
 
     try {
-      await updateAndRefetchOrderForm({
-        variables: {
-          orderFormId: orderForm.orderFormId,
-          items: updatedItem,
-        },
-      })
+      await updateItems(updatedItems)
```

Comment:
Shouldn't you add a loading also?

Reply:
Why should I? The item is removed from the minicart. This is the background sync magic :sparkles: